CI is failing - integration tests - php8 Required parameter $id_shop follows optional parameter $id_linksmenutop

#### Describe the bug

CI is failing on Integration tests with PHP8
```
--- Failed steps:

001 Scenario: Create a simple customer and edit it                         # Features/Scenario/Customer/customer_management.feature:11
      When I edit customer "CUST-1" and I change the following properties: # Features/Scenario/Customer/customer_management.feature:21
        8192: Required parameter $id follows optional parameter $token in /home/runner/work/PrestaShop/PrestaShop/modules/ps_emailsubscription/ps_emailsubscription.php line 316

002 Scenario: Edit category                                        # Features/Scenario/Category/category_management.feature:15
      Given I add new category "category1" with following details: # Features/Scenario/Category/category_management.feature:9
        8192: Required parameter $id_shop follows optional parameter $id_linksmenutop in /home/runner/work/PrestaShop/PrestaShop/modules/ps_mainmenu/ps_menutoplinks.class.php line 29

808 scenarios (806 passed, 2 failed)
13734 steps (13729 passed, 2 failed, 3 skipped)
21m39.28s (343.93Mb)

Script @php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict handling the integration-behaviour-tests event returned with error code 1``

Error: Process completed with exit code 1.
```

#### Expected behavior

CI should be green ... 😆 

#### Steps to Reproduce

See Continuous Integration report from any PR

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -54,7 +54,7 @@ jobs:
               restore-keys: ${{ runner.os }}-composer-
 
           - name: Composer Install
-            run: composer install --ansi --prefer-dist --no-interaction --no-progress --quiet
+            run: composer install --ansi --prefer-dist --no-interaction --no-progress
 
           - name: Build assets
             run: make assets

--- a/composer.json
+++ b/composer.json
@@ -195,7 +195,7 @@
         "git-hook-install": "@php .github/contrib/install.php",
         "git-hook-uninstall": "@php .github/contrib/uninstall.php",
         "integration-behaviour-tests": [
-            "@php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict --stop-on-failure"
+            "@php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict"
         ],
         "integration-tests": [
             "@composer create-test-db",

--- a/composer.json
+++ b/composer.json
@@ -77,7 +77,7 @@
         "prestashop/ps_customersignin": "^2",
         "prestashop/ps_customtext": "^4",
         "prestashop/ps_dataprivacy": "^2.0",
-        "prestashop/ps_emailsubscription": "^2.5",
+        "prestashop/ps_emailsubscription": "^2.7",
         "prestashop/ps_facetedsearch": "^3.2.1",
         "prestashop/ps_faviconnotificationbo": "^2",
         "prestashop/ps_featuredproducts": "^2",

--- a/composer.json
+++ b/composer.json
@@ -195,7 +195,7 @@
         "git-hook-install": "@php .github/contrib/install.php",
         "git-hook-uninstall": "@php .github/contrib/uninstall.php",
         "integration-behaviour-tests": [
-            "@php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict"
+            "@php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict --stop-on-failure"
         ],
         "integration-tests": [
             "@composer create-test-db",

--- a/src/Core/Addon/Module/ModuleManager.php
+++ b/src/Core/Addon/Module/ModuleManager.php
@@ -287,10 +287,6 @@ class ModuleManager implements AddonManagerInterface
             $source = null;
         }
 
-        if ($this->moduleProvider->isInstalled($name)) {
-            return $this->upgrade($name, 'latest', $source);
-        }
-
         if (!empty($source)) {
             $this->moduleZipManager->storeInModulesFolder($source);
         } elseif (!$this->moduleProvider->isOnDisk($name)) {

--- a/src/Core/Addon/Module/ModuleManager.php
+++ b/src/Core/Addon/Module/ModuleManager.php
@@ -287,6 +287,15 @@ class ModuleManager implements AddonManagerInterface
             $source = null;
         }
 
+        if ($this->moduleProvider->isInstalled($name)) {
+            if ($source !== null) {
+                return $this->upgrade($name, 'latest', $source);
+            }
+
+            // Module is already installed
+            return true;
+        }
+
         if (!empty($source)) {
             $this->moduleZipManager->storeInModulesFolder($source);
         } elseif (!$this->moduleProvider->isOnDisk($name)) {

